pandas/tests/frame/test_reshape.py

```diff
@@ -918,6 +918,90 @@ def test_unstack_swaplevel_sortlevel(self, level):
         tm.assert_frame_equal(result, expected)
 
 
+def test_explode():
```

Comment:
Can you parametrize this test instead?

Reply:
That's usually my default preference but for this case it would mean putting a ton of things in a `@pytest.mark.parametrize` decorator, which didn't smell right. Maybe I'm missing a better pattern to parametrize though, can you point me to an example you think I should follow here?

Reply:
ok, so then this is more about splitting it out into multiple tests rather than parametrize right?

Reply:
ok, I split the test cases out and just pushed an update